website/source/docs/guides/production-acls.html.md

```diff
@@ -0,0 +1,484 @@
+ } 
+```
+
+If the service is running, you will need to restart it. Unlike agent tokens,
+service tokens must be set in the configuration file; you cannot set them with
+the HTTP API. However, if you register a service with the API, you can pass the
+token in the [header](https://www.consul.io/api/index.html#authentication) with `X-Consul-Token` and it will be used by the service.
+
+If you are using a sidecar proxy, it can inherit the token from the service
+definition. Alternatively, you can create a separate token. 
+
+## Token for DNS
+
+Depending on your use case, the token used for DNS may need policy rules for
+[nodes](https://www.consul.io/docs/agent/acl-rules.html#node-rules),
+[services](https://www.consul.io/docs/agent/acl-rules.html#service-rules), and
+[prepared queries](https://www.consul.io/docs/agent/acl-rules.html#prepared-query-rules).
+You should apply the token to the Consul agent serving DNS requests. When the
+DNS server makes a request to Consul, it will include the token in the request.
+Consul can either authorize or revoke the request, depending on the token's
+privileges. The token creation for DNS is the same three step process you used
+for agents and services, create a policy, create a token, apply the
+token. 
+
+Below is an example of a policy that provides read privileges for all services,
+nodes, and prepared queries. 
+
+```
+# dns-request-policy.hcl
+node_prefix "" { 
+	policy = "read" 
+} 
+service_prefix "" { 
+	policy = "read" 
+}
+# only needed if using prepared queries
+query_prefix "" { 
+	policy = "read" 
+} 
+```
+
+First, create the policy.
+
+``` 
+$ consul acl policy create -name "dns-requests" -rules @dns-request-policy.hcl 
+```
+
+Next, create the token.
+
+``` 
+$ consul acl token create -description "Token for DNS Requests" -policy-name dns-requests 
+```
+
+Finally, apply the token to the Consul agent serving DNS request in default token ACL
+configuration parameter.
+
+``` 
+{ 
+  "acl": { 
+  	"enabled": true, 
+  	"default_policy": "deny", 
+  	"tokens": { 
+  		"agent": "e67a4a33-6579-b766-ac97-a0408fde3957", 
+  		"default":"5467d69a-5f19-469b-0543-12a487eecc66", 
+  	}, 
+  },
+} 
+``` 
+
+## Consul KV Tokens
+
+The  process of creating tokens for Consul KV follows the same three step
+process as nodes and services. First create a policy, then a token, and finally
+apply or use the token. However, unlike tokens for nodes and services Consul KV
+has many varied use cases. 
+
+- Services may need to access configuration data in the key-value store. 
+- You may want to store distributed lock information for sessions.  
+- Operators may need access to
+update configuration values in the key-value store. . 
+
+The [rules for
+KV](https://www.consul.io/docs/agent/acl-rules.html#key-value-rules) have four
+policy levels; `deny`, `write`, `read`, and `list`.  Let's review several
+examples of `read` and `write`.
+
+Depending on the use case, the token will be applied differently. For services
```

Comment:
This isn't really accurate.

Consul agents in and of themselves will not initiate kv reads or writes. This only happens on behalf of some HTTP client. The tokens need to be given to whatever those HTTP clients are and then used in requests being made back to Consul. The token within the service definition will never be used for kv related operations (unless that token is being shared with another process that makes those HTTP requests).

Reply:
Makes sense.